ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -668,4 +693,60 @@ public JoinKey rewriteWith(
       return this;
     }
   }
+
+  private static final class ForeignJoinKey implements JoinKey {
+    private final ColumnName foreignKeyColumn;
```

Comment:
Why do we include this variable? The purpose of `JoinKey` is to define the key of the result table, and thus it seem it does not need to know anything about the foreign key?